specification/chaos/resource-manager/Microsoft.Chaos/preview/2023-04-15-preview/types/experiments.json

```diff
@@ -0,0 +1,867 @@
+    "discreteAction": {
+      "type": "object",
+      "description": "Model that represents a discrete action.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/action"
+        }
+      ],
+      "properties": {
+        "parameters": {
+          "description": "List of key value pairs.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/keyValuePair"
+          },
+          "x-ms-identifiers": [
+            "key"
+          ]
+        },
+        "selectorId": {
+          "type": "string",
+          "description": "String that represents a selector.",
+          "minLength": 1
+        }
+      },
+      "required": [
+        "parameters",
+        "selectorId"
+      ],
+      "additionalProperties": false,
+      "x-ms-discriminator-value": "discrete"
+    },
+    "continuousAction": {
+      "type": "object",
+      "description": "Model that represents a continuous action.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/action"
+        }
+      ],
+      "properties": {
+        "duration": {
+          "type": "string",
+          "description": "ISO8601 formatted string that represents a duration.",
+          "pattern": "^P(\\d+Y)?(\\d+M)?(\\d+D)?(T(\\d+H)?(\\d+M)?(\\d+(\\.\\d+)?S)?)?$",
+          "format": "duration"
+        },
+        "parameters": {
+          "description": "List of key value pairs.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/keyValuePair"
+          },
+          "x-ms-identifiers": [
+            "key"
+          ]
+        },
+        "selectorId": {
+          "type": "string",
+          "description": "String that represents a selector.",
+          "minLength": 1
+        }
+      },
+      "required": [
+        "duration",
+        "parameters",
+        "selectorId"
+      ],
+      "additionalProperties": false,
+      "x-ms-discriminator-value": "continuous"
+    },
+    "keyValuePair": {
+      "description": "A map to describe the settings of an action.",
+      "type": "object",
+      "properties": {
+        "key": {
+          "description": "The name of the setting for the action.",
+          "type": "string",
+          "minLength": 1
+        },
+        "value": {
+          "description": "The value of the setting for the action.",
+          "type": "string",
+          "minLength": 1
+        }
+      },
+      "required": [
+        "key",
+        "value"
+      ],
+      "additionalProperties": false
+    },
+    "selector": {
+      "type": "object",
+      "description": "Model that represents a selector in the Experiment resource.",
+      "discriminator": "type",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Enum of the selector type.",
+          "enum": [
+            "List",
+            "Query"
+          ],
+          "x-ms-enum": {
+            "name": "SelectorType",
+            "modelAsString": false
```

Comment:
>false [](http://example.com/codeflow?start=29&length=5)

Why false for this? This can break client if you need to update the enum

Reply:
Thanks

Reply:
No explicit reason, string works for us - will change.

Reply:
changes applied